File: examples/scrape-structured-data/rest/csharp/ScrapeStructuredData.cs
// Extracts structured data from a page using the Browserless /scrape endpoint.
// Uses System.Net.Http and System.Text.Json from the .NET standard library.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

const string Token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/scrape?token={Token}";

var payload = new
{
    url = "https://example.com",
    elements = new[] { new { selector = "h1" }, new { selector = "p" } },
};

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, endpoint);
request.Content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");

var response = await client.SendAsync(request);
string responseBody = await response.Content.ReadAsStringAsync();
Console.WriteLine(responseBody);

File: examples/scrape-structured-data/rest/java/ScrapeStructuredData.java
// Extracts structured data from a page using the Browserless /scrape endpoint.
// java.net.http is included in JDK 11+ — no extra dependency needed.
//
// Run: javac ScrapeStructuredData.java && java ScrapeStructuredData

import java.net.URI;
import java.net.http.*;

public class ScrapeStructuredData {
    private static final String TOKEN = "YOUR_API_TOKEN_HERE";

    public static void main(String[] args) throws Exception {
        String endpoint = "https://production-sfo.browserless.io/scrape?token=" + TOKEN;
        String jsonData = """
            {
                "url": "https://scraping-sandbox.netlify.app/products",
                "elements": [
                    { "selector": "h1" },
                    { "selector": "p" }
                ]
            }""";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonData))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response.body());
    }
}
